src/fetch/transfer_file.py

```diff
@@ -141,7 +159,19 @@ def erc20(
     @property
     def amount(self) -> float:
         """Returns transfer amount in units"""
-        return self.amount_wei / 10**18
+        return self.amount_wei / 10 ** self.token_decimals()
+
+    def token_decimals(self) -> int:
+        """Fetches token decimals from EVM when not already known."""
+        if self.token_type == TokenType.NATIVE:
+            return 18
+        try:
+            return cached_token_decimals(self.token_address)
+        except web3.exceptions.BadFunctionCallOutput as err:
+            logging.error(
+                f"failed to fetch token decimals for {self.token_address}. returning 18"
```

Comment:
Is this needed for us? Isn't it saver to just throw an error?

Reply:
If we throw an error we can't unit test with random token addresses. I proposed an alternative option to this solution in the PR description.